### DISCUSSION TITLE
Extend site_meta with parametrized values stored in the docs folder

### DISCUSSION THREAD

**Discussion by 9Lukas5** [squidfunk/mkdocs-material#4614] [Q&A]
Hey, I'm not sure if I'm missing something I can't see or it's just not possible, what I'm trying to do

My Folder structure:

```
docs
└── title-of-page
    ├── text.md
    └── values.jinja
overrides
└── main.html
```

What I'm trying to do is, I want some more things in the site_meta, for which I made the main.html override.
The things I wanna add should be individual for each page I make, stored in the `values.jinja` file right beside it.

So what I tried, is this:

```html
{% extends "base.html" %}

{% block site_meta %}
  {{ super() }}

  {% set valuesFile = page.url | replace("text.html", "values.jinja") %}
  {% set values = dict() %}
  {% from valuesFile import values %}

  {% if values.title %}
    <meta property="og:title" content="{{ values.title }}">
  {% endif %}
  {% if values.description %}
    <meta property="og:description" content="{{ values.description }}">
  {% endif %}

{% endblock %}
```

But it always fails with `TemplateNotFound`.
Something's not right with the attempt to grab the right path anyway, because if I manually set a string for `valuesFile` the error message also includes the Template I tried to load, which is just empty if I try it from the variable.

So I have basically two questions:
- Can I import something from the `docs/` directory in an override?
- If so, what's the best way to extract the path of the values file?

**Comment by squidfunk** [comment]
In order to override templates, you must go through the [theme extension](https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme) process. You can't put templates into `docs`. If you want to define data individual to each page, [front matter is your best option](https://squidfunk.github.io/mkdocs-material/reference/#using-metadata-in-templates), which you can use inside templates with `page.meta.*`.

**Reply by 9Lukas5** [reply]
Mhhhh, ok I see how this would be accessible with the information from the last link.

Thing is, I constructed this :nerd_face: :

```
docs
├── pages
│   └── title-of-page
│       ├── text.md
│       └── values.jinja
└── index.md
```
index.md:

```markdown
{% for entry in config.nav %}
{% if entry is defined %}
{% for entry in entry.pages %}
    {% set markdownFile = entry.values() | list %}
    {% set markdownFile = markdownFile[0] %}
    {% set valuesFile = markdownFile | replace("text.md", "values.jinja") %}
    {% import valuesFile as post %}
## [{{ post.values.title }}]({{ markdownFile }})

<p class="introduction" markdown>
<span class="date">{{ post.values.date }}</span>  
{{ post.values.description }}
</p>

---

{% endfor %}
{% endif %}
{% endfor %}
```

This lets every page automatically popup on the index page with its title, date and description.
I put these values into a separate file and imported it in the index file as well as the dedicated page file, as the `page.*` variable wouldn't be accessible from the `index.md`.

And in the page YAML header where I can set the meta variables, which would be accessible from the theme template, I were not able to use `{{ variable }}` :thinking: 

Any idea how I can write the description once, and have it accesible on a per page basis

- in the page in question
- from another page
- in the theme templates

:eyes:
